### ISSUE TITLE
Property does not exist on type in admin

### ISSUE THREAD

**Issue by bot101** [ever-co/ever#968]
I have faced a number of issues while trying to setup this project, especially with the symlinks and node-sass. I have resolved all of those successfully but I am unable to resolve this issue with properties not eisting on a number of objects. This error is from an elevated Command Line interface on Windows 10. The same errors also show up in VSCode intellisense. Please advice on how to fix, thanks.

```
ERROR in app/@core/data/device.service.ts:115:46 - error TS2339: Property 'update' does not exist on type 'IDeviceRawObject'.

115                     map((result) => result.data.updateDevice.update),
                                                                 ~~~~~~
app/@core/data/device.service.ts:134:33 - error TS2339: Property 'removeDeviceByIds' does not exist on type '{}'.

134                     map((result) => result.data.removeDeviceByIds),
                                                    ~~~~~~~~~~~~~~~~~
app/@core/data/device.service.ts:140:3 - error TS2322: Type 'Observable<IDeviceRawObject>' is not assignable to type 'Observable<Device>'.
  Type 'IDeviceRawObject' is missing the following properties from type 'Device': isDeleted, CreateObjectTYPE, RawObjectTYPE, createdAt, and 3 more.

140     return this.apollo
        ~~~~~~~~~~~~~~~~~~
141             .mutate<{ createDevice: IDeviceRawObject }>({
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
155                     share()
    ~~~~~~~~~~~
156             );
    ~~~~~
app/@core/data/products.service.ts:101:33 - error TS2339: Property 'createProduct' does not exist on type '{ product: IProductCreateObject; }'.

101                     map((result) => result.data.createProduct),
                                                    ~~~~~~~~~~~~~
app/@core/data/products.service.ts:121:33 - error TS2339: Property 'saveProduct' does not exist on type '{ product: Product; }'.

121                     map((result) => result.data.saveProduct),
                                                    ~~~~~~~~~~~
app/@core/data/products.service.ts:139:33 - error TS2339: Property 'removeProductsByIds' does not exist on type '{}'.

139                     map((result) => result.data.removeProductsByIds),
                                                    ~~~~~~~~~~~~~~~~~~~
app/@core/data/productsCategory.service.ts:66:33 - error TS2339: Property 'createProductsCategory' does not exist on type '{ productsCategory: IProductsCategoryCreateObject; }'.

66                      map((result) => result.data.createProductsCategory),
                                                    ~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/productsCategory.service.ts:105:33 - error TS2339: Property 'updateProductsCategory' does not exist on type '{ id: string; productsCategory: IProductsCategoryCreateObject; }'.

105                     map((result) => result.data.updateProductsCategory),
                                                    ~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/warehouses.service.ts:220:33 - error TS2339: Property 'warehouseAddProducts' does not exist on type '{ warehouseId: string; products: IWarehouseProduct[]; }'.

220                     map((result) => result.data.warehouseAddProducts),
                                                    ~~~~~~~~~~~~~~~~~~~~
app/@core/data/warehouseOrders.service.ts:39:33 - error TS2339: Property 'createOrder' does not exist on type '{}'.

39              .map((result) => result.data.createOrder);
                                             ~~~~~~~~~~~
app/@core/auth/auth.module.ts:6:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

6 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@theme/components/footer/footer.component.ts:2:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

2 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/admins.service.ts:78:33 - error TS2339: Property 'updateAdmin' does not exist on type '{ id: string; updateInput: IAdminUpdateObject; }'.

78                      map((result) => result.data.updateAdmin),
                                                    ~~~~~~~~~~~
app/app.module.ts:25:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

25 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
main.ts:11:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

11 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/invites.service.ts:104:33 - error TS2339: Property 'createInvite' does not exist on type '{ createInput: IInviteCreateObject; }'.

104                     map((result) => result.data.createInvite),
                                                    ~~~~~~~~~~~~
app/@core/data/invites.service.ts:122:33 - error TS2339: Property 'removeInvitesByIds' does not exist on type '{}'.

122                     map((result) => result.data.removeInvitesByIds),
                                                    ~~~~~~~~~~~~~~~~~~
app/@core/data/invites.service.ts:149:33 - error TS2339: Property 'updateInvite' does not exist on type '{ id: string; updateInput: IInviteUpdateObject; }'.

149                     map((result) => result.data.updateInvite),
                                                    ~~~~~~~~~~~~
app/@core/data/invites-requests.service.ts:119:33 - error TS2339: Property 'createInviteRequest' does not exist on type '{ createInput: IInviteRequestCreateObject; }'.

119                     map((result) => result.data.createInviteRequest),
                                                    ~~~~~~~~~~~~~~~~~~~
app/@core/data/invites-requests.service.ts:137:33 - error TS2339: Property 'removeInvitesRequestsByIds' does not exist on type '{}'.

137                     map((result) => result.data.removeInvitesRequestsByIds),
                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/invites-requests.service.ts:167:33 - error TS2339: Property 'updateInviteRequest' does not exist on type '{ id: string; updateInput: IInviteRequestUpdateObject; }'.

167                     map((result) => result.data.updateInviteRequest),
                                                    ~~~~~~~~~~~~~~~~~~~
app/@core/data/fakeDataServices/carriers.ts:6:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

6 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/fakeDataServices/invites.ts:5:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

5 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/fakeDataServices/storageService.ts:6:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

6 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/fakeDataServices/users.ts:7:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

7 import { environment } from '../../../../environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/data/fakeDataServices/warehouses.ts:7:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

7 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@core/utils/passwords.service.ts:2:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

2 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@shared/file-uploader/file-uploader.component.ts:9:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

9 import { environment } from 'environment';
                              ~~~~~~~~~~~~~
app/@shared/forms/google-map/google-map.module.ts:4:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

4 import { environment } from 'environment';
                              ~~~~~~~~~~~~~
app/pages/+carriers/+carrier/location/carrier-location.component.ts:9:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

9 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@shared/order/order-map/user-warehouse-map/user-warehouse-map.ts:12:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

12 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@shared/order/order-map/carreir-location/carreir-location.ts:10:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

10 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/@shared/render-component/store-products-table/store-product-price.component.ts:3:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

3 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/+carriers/track/carrier-tracking/carrier-tracking.component.ts:15:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

15 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/+carriers/+carrier/location/carrier-location.module.ts:9:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

9 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/+carriers/+carrier/carrier-map/carrier-map.component.ts:13:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

13 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/+fakeData/fakeData.component.ts:44:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

44 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/+maintenance-info/maintenance-info.component.ts:3:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

3 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/+warehouses/+warehouse/warehouse-order-view/warehouse-order-view.component.ts:17:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

17 import { environment } from 'environments/environment';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/+server-down/server-down.page.ts:5:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

5 import { environment } from 'environment';
                              ~~~~~~~~~~~~~
app/pages/+warehouses/+warehouse-track/warehouse-track.component.ts:3:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

3 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~
app/pages/maps/maps.module.ts:7:29 - error TS2306: File 'C:/Users/PC/Desktop/ever/admin/website-angular/src/environments/environment.ts' is not a module.

7 import { environment } from 'environments/environment';
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~

ERROR in Error during template compile of 'AppModule'
  Function calls are not supported in decorators but 'ThemeModule' was called.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

**Comment by evereq** [comment]
Hi @bot101, how exactly you try to run Admin UI project? It looks like many errors related to environment.ts can't be found. We have such file automatically generated when you run project using `yarn run:admin` (see our README how to run each project in the platform). More specifically, it executes following command (see package.json file in the root of the repo):
```
yarn --cwd ./admin/website-angular start
```
And effect of command above is actually run for another command (from https://github.com/ever-co/ever/blob/develop/admin/website-angular/package.json#L43): 
```
yarn run config:dev && ng serve
```
As you can see the first we run `config:dev`, which is actually (see https://github.com/ever-co/ever/blob/develop/admin/website-angular/package.json#L41):
```
yarn run config -- --environment=dev
```
and that actually runs (see https://github.com/ever-co/ever/blob/develop/admin/website-angular/package.json#L40)
```
yarn ts-node ./scripts/configure.ts
```
and configure.ts generates automatically environment.ts file with setting defined in .env file (or default if you don't have .env.ts file at all)

To summarize: at least most of your errors seem because for some reason environment.ts file is not generated, which can be because you did not run startup scripts correct way.

I hope that helps!

**Comment by evereq** [comment]
Note: we did some changes recently which related to how we build environment.ts files, so that should help to resolve such issues, closing a ticket.
